src/Storages/MergeTree/DataPartsExchange.cpp

```diff
@@ -52,6 +52,24 @@ std::string getEndpointId(const std::string & node_id)
     return "DataPartsExchange:" + node_id;
 }
 
+/// Simple functor for tracking fetch progress in system.replicated_fetches table.
+struct ReplicatedFetchReadCallback
+{
+    ReplicatedFetchList::Entry & replicated_fetch_entry;
+
+    explicit ReplicatedFetchReadCallback(ReplicatedFetchList::Entry & replicated_fetch_entry_)
+        : replicated_fetch_entry(replicated_fetch_entry_)
+    {}
+
+    void operator() (size_t bytes_count)
+    {
+        replicated_fetch_entry->bytes_read_compressed.store(bytes_count, std::memory_order_relaxed);
+        replicated_fetch_entry->progress.store(
+                static_cast<double>(bytes_count) / replicated_fetch_entry->total_size_bytes_compressed,
```

Comment:
Is it possible that `total_size_bytes_compressed` is 0? I see that initial value is 0:
https://github.com/ClickHouse/ClickHouse/blob/1c9e24b5e2ab5c753b71a66e5d0d7c86e7df401c/src/Storages/MergeTree/DataPartsExchange.cpp#L285

Reply:
Wow, yes. Of course, it's not possible, but initialization should be fixed.